app/src/main/kotlin/com/simplemobiletools/keyboard/views/MyKeyboardView.kt

```diff
@@ -1390,12 +1390,15 @@ class MyKeyboardView @JvmOverloads constructor(context: Context, attrs: Attribut
                 }
                 showPreview(NOT_A_KEY)
                 Arrays.fill(mKeyIndices, NOT_A_KEY)
+
+                val currentKeyCode = mKeys.getOrNull(mCurrentKey)?.code
+
                 // If we're not on a repeating key (which sends on a DOWN event)
                 if (mRepeatKeyIndex == NOT_A_KEY && !mMiniKeyboardOnScreen && !mAbortKey) {
                     detectAndSendKey(mCurrentKey, touchX, touchY, eventTime)
                 }
 
-                if (mKeys.getOrNull(mCurrentKey)?.code == KEYCODE_SPACE && !mIsLongPressingSpace) {
+                if (currentKeyCode == KEYCODE_SPACE && !mIsLongPressingSpace) {
                     detectAndSendKey(mCurrentKey, touchX, touchY, eventTime)
                 }
```

Comment:
By the way, I don't like this place. Maybe we should use if ... else ... instead of two if statememnts

Reply:
In last commit there was a situation when you push on a change layout button, then it changes from the first if statement and immediately detectAndSendKey emits KEYCODE_SPACE which changes layout again

Reply:
what do you mean? How to reproduce some glitch?

Reply:
Try changing the layout on the Danish keyboard, for example. Letters layout remains active

Reply:
right, so that is a bug in this part? Why does switching to symbols in other languages work and in danish it just enters a space?

Reply:
Yep, investigating now

Reply:
> right, so that is a bug in this part? Why does switching to symbols in other languages work and in danish it just enters a space?

Got it. It happens with Scandinavian keyboards because of their layout. They all have additional 11th letter in the center row. The bug processes next way: 

1. User clicks on change layout button
2. First if statement performs and it changes keyboard layout and it's keys (symbol keys). Keys arrayList value of the keyboard is updated and now it contains symbols (not letters)
3. Then, on the second if statement, `mKeys.getOrNull(mCurrentKey)?.code == KEYCODE_SPACE` the mKeys already takes another button by mCurrentKey, as it's arrayList was updated and the sane mCurrent key index is pointing at the space bar, which specifies this if as true and sends the space button click.

So we need to remember the key before we initialize new keys to protect ourselves against errors.